### DISCUSSION TITLE
Requires SystemD

### DISCUSSION THREAD

**Discussion by h-2** [oskarsh/Yin-Yang#247] [Q&A]
I am trying to build it, but it fails:

```
Collecting systemd-python==234
  Using cached systemd-python-234.tar.gz (53 kB)
  Preparing metadata (setup.py) ... error
  error: subprocess-exited-with-error
  
  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [12 lines of output]
      Cannot find libsystemd or libsystemd-journal:
      
      Package libsystemd was not found in the pkg-config search path.
      Perhaps you should add the directory containing `libsystemd.pc'
      to the PKG_CONFIG_PATH environment variable
      No package 'libsystemd' found
      
      Package libsystemd-journal was not found in the pkg-config search path.
      Perhaps you should add the directory containing `libsystemd-journal.pc'
      to the PKG_CONFIG_PATH environment variable
      No package 'libsystemd-journal' found
      
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
```

Does this require SystemD? Can I work around it?

**Comment by l0drex** [comment]
Yes, we use systemd for the scheduling. You need systemd and it's developer dependencies

**Comment by l0drex** [comment]
I you don't want systemd, you may be able to work around it:
- remove the systemd-python dependency from `requirements.txt`
- run the installation script
- don't enable automatic switching in the gui
- create a cronjob to execute `yin-yang --toggle` to switch the theme

**Comment by h-2** [comment]
> I you don't want systemd, you may be able to work around it:

OK, I'll give that a try, thanks! [My Distro doesn't have SystemD]

> create a cronjob to execute yin-yang --toggle to switch the theme

I would probably just bind `yin-yang --toggle` to a keyboard shortcut anyway.

**Comment by l0drex** [comment]
> I would probably just bind `yin-yang --toggle` to a keyboard shortcut anyway.

There is also a system tray icon in the latest beta version with a toggle option in the right click menu.

**Comment by beedell-roke** [comment]
This is probably best moved to a discussion considering it's more a support request than an issue, right?
